### PR TITLE
Implement `plane_to_sphere`

### PR DESCRIPTION
**Pull request by milancurcic**

### CODE REVIEW

File: clouddrift/sphere.py [hunk 1]
```diff
@@ -105,6 +105,66 @@ def recast_lon180(lon: np.ndarray) -> np.ndarray:
...
+    Raises
+    ------
+    AttributeError
+        If ``x`` and ``y`` are not NumPy arrays
+
+    See Also
+    --------
+    :func:`sphere_to_plane`
+    """
+    lon = np.empty(x.shape, dtype=np.float64)
```
Comment (philippemiron) [inline clouddrift/sphere.py:148]: You could use `np.empty_like` to keep the type of the input.

File: clouddrift/sphere.py [hunk 2]
```diff
@@ -143,16 +203,20 @@ def sphere_to_plane(
...
     ------
-    TypeError
+    AttributeError
         If ``lon`` and ``lat`` are not NumPy arrays
+
+    See Also
+    --------
+    :func:`plane_to_sphere`
     """
     x = np.empty(lon.shape, dtype=np.float64)
```
Comment (philippemiron) [inline clouddrift/sphere.py:221]: Same here

**Review by philippemiron — APPROVED** [review]

File: clouddrift/sphere.py [hunk 3]
```diff
@@ -105,6 +105,66 @@ def recast_lon180(lon: np.ndarray) -> np.ndarray:
...
+    ------
+    AttributeError
+        If ``x`` and ``y`` are not NumPy arrays
+
+    See Also
+    --------
+    :func:`sphere_to_plane`
+    """
+    lon = np.empty(x.shape, dtype=np.float64)
+    lat = np.empty(y.shape, dtype=np.float64)
```
Comment (philippemiron) [inline clouddrift/sphere.py:149]:
```suggestion
    lon = np.empty_like(x)
    lat = np.empty_like(y)
```

File: clouddrift/sphere.py [hunk 4]
```diff
@@ -143,16 +203,20 @@ def sphere_to_plane(
...
-    TypeError
+    AttributeError
         If ``lon`` and ``lat`` are not NumPy arrays
+
+    See Also
+    --------
+    :func:`plane_to_sphere`
     """
     x = np.empty(lon.shape, dtype=np.float64)
     y = np.empty(lat.shape, dtype=np.float64)
```
Comment (philippemiron) [inline clouddrift/sphere.py:222]:
```suggestion
     x = np.empty_like(lon)
     y = np.empty_like(lat)
```

**Review by philippemiron — APPROVED** [review]

**Comment by philippemiron** [comment]
By default in python (and most people use like this since you don't have to define type of course) everything is in `float64`. 

The reason my suggestion would be to keep `type` is that if it's `float32`, it's most likely because the user switched the type to save space, for compatibility with another app (pytorch for example), etc., so switching to 64 bits might create confusion having inputs/outputs of different types.

We can add a note that round-trip of position on the Earth are not exact if precision is decreased. But at the same time, this is almost always the case for any function. 

For example:

**Comment by milancurcic** [comment]
Thanks for the review.

**Pull request merged**